server/app/views/admin/programs/ProgramBlockAllAspectsView.java

```diff
@@ -397,21 +394,67 @@ private DivTag blockEditPanel(
                       blockQuestions.size()));
             });
 
-    ButtonTag addQuestion =
+    DivTag div = div().withClasses("w-7/12", "py-6", "px-4");
+
+    // UI elements for editing are only needed when we view a draft
+    if (viewAllowsEditingProgram()) {
+      DivTag buttons = blockPanelButtons(program, blockDefinitionIsEnumerator, blockDescriptionModalButton, canDelete);
+      ButtonTag addQuestion =
         makeSvgTextButton("Add a question", Icons.ADD)
-            .withClasses(
-                AdminStyles.PRIMARY_BUTTON_STYLES,
-                ReferenceClasses.OPEN_QUESTION_BANK_BUTTON,
-                "my-4");
+          .withClasses(
+            AdminStyles.PRIMARY_BUTTON_STYLES,
+            ReferenceClasses.OPEN_QUESTION_BANK_BUTTON,
+            "my-4");
+
+      div.with(blockInfoDisplay, buttons, visibilityPredicateDisplay);
+      maybeEligibilityPredicateDisplay.ifPresent(div::with);
+      return div.with(programQuestions, addQuestion);
+    } else {
+      div.with(blockInfoDisplay, visibilityPredicateDisplay);
+      maybeEligibilityPredicateDisplay.ifPresent(div::with);
+      return div.with(programQuestions);
+    }
+  }
 
-    DivTag div =
-        div()
-            .withClasses("w-7/12", "py-6", "px-4")
-            .with(blockInfoDisplay, buttons, visibilityPredicateDisplay);
-    maybeEligibilityPredicateDisplay.ifPresent(div::with);
+  private DivTag blockPanelButtons(
+    ProgramDefinition program,
+    boolean blockDefinitionIsEnumerator,
+    ButtonTag blockDescriptionModalButton,
+    boolean canDelete){
 
-    return div.with(programQuestions, addQuestion);
-  }
+    // Add buttons to change the block.
+    DivTag buttons = div().withClasses("flex", "flex-row", "gap-4");
+
+    // Buttons are only needed when the view is used for editing
+      buttons.with(blockDescriptionModalButton);
+      if (blockDefinitionIsEnumerator) {
+        buttons.with(
```

Comment:
You can bake the if into this condWith

buttons.condWith(blockDefinitionIsEnumerator, button(".......